src/ExtensionSet.php

```diff
@@ -437,17 +435,26 @@ private function initExtension(ExtensionInterface $extension): void
     {
         // filters
         foreach ($extension->getFilters() as $filter) {
-            $this->filters[$filter->getName()] = $filter;
+            $this->filters[$name = $filter->getName()] = $filter;
+            if (str_contains($name, '*')) {
+                $this->dynamicFilters['#^'.str_replace('\\*', '(.*?)', preg_quote($name, '#')).'$#'] = $filter;
```

Comment:
Small optimization here to avoid doing the same work over and over again. And having just the dynamic filters in an array means looping only on them.